scheduler/filter.go

```diff
@@ -0,0 +1,100 @@
+package scheduler
+
+import (
+	"github.com/Sirupsen/logrus"
+	"sort"
+)
+
+// Filter is an interface to prioritize and filter the hosts
+type Filter interface {
+	Filter(scheduler *Scheduler, resourceRequest []ResourceRequest, context Context, hosts []string) []string
+}
+
+// ProvideFilter provides a list of filter. It provides a gateway to register a new filter.
+func ProvideFilter() []Filter {
```

Comment:
My comment about the sounding of ProvideReserveAction applies to this method name and to ProvideReleaseAction as well. Can you follow the same naming convention?